glue/viewers/profile/state.py

```diff
@@ -280,7 +316,13 @@ def update_profile(self, update_limits=True):
             data = self.layer
             subset_state = None
 
-        profile_values = data.compute_statistic(self.viewer_state.function, self.attribute, axis=axes, subset_state=subset_state)
+        if self.viewer_state.function == 'slice':
+            data_slice = list(copy.deepcopy(self.viewer_state.slices))
+            data_slice[pix_cid.axis] = slice(None)
+            profile_values = data.get_data(self.attribute, view=tuple(data_slice))
```

Comment:
For subsets you might also need to call ``get_mask`` and then blank out any non-selected values in the profile (see also my comment about subsets in the main review)

Reply:
Sure